lib/graphql/relay/walker/query_builder.rb

```diff
@@ -106,7 +106,11 @@ def field_ast(field, arguments={}, &blk)
 
       # Bail unless we have the required arguments.
       return unless field.arguments.reject do |_, arg|
-        valid_input?(arg.type, nil)
+        if GraphQL::VERSION >= "1.5.6"
+          type.valid_isolated_input?(arg.type)
+        else
+          valid_input?(arg.type)
+        end
```

Comment:
I think this part can remain unchanged since the later switch on `GraphQL::VERSION` provides the right implementation of `valid_input?`